paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/CdcActionCommonUtils.java

```diff
@@ -205,7 +210,7 @@ private static void setPrimaryKeys(
             }
         }
 
-        if (!sourceSchemaPrimaryKeys.isEmpty()) {
+        if (usePKeysFromSourceForPaimonSchema && !sourceSchemaPrimaryKeys.isEmpty()) {
```

Comment:
Can you add a test case for this PR?

Reply:
The only tests that I find for this are IT tests. I can add them. Do you want me to add them for all CDC sources or just adding one and verifying it is enough?